uae4all_gp2x_0.7.2a/src/od-joy.mm

```diff
@@ -116,6 +152,27 @@ void read_joystick(int nr, unsigned int *dir, int *button)
     if (right) bot = !bot;
     *dir = bot | (right << 1) | (top << 8) | (left << 9);
 #endif
+    if(mainViewController == nil)
+    {}
+    else if(mainMenu_servermode>1) //am I the client, then I do should send notifications to the server
+    {
+        mainMenu_joy1button = *button;
+        mainMenu_joy1dir = *dir;
+        sendJoystickDataToServer (mainViewController, *dir);
```

Comment:
@simontoens Hi simon,
It is also used as a parameter in the call sendJoystickDataToServer (mainViewController, *dir);
I had first some difficulties to make a call from this .mm class to the specific object instance of the controller in MainEmulationViewController.m, which is in charge of sending the collected data to the server. I found this wrapping approach which worked fine, where you use the controller object as an parameter.